asreview/data.py

```diff
@@ -311,15 +312,21 @@ def record(self, i, by_index=True):
 
         if not by_index:
             records = [
-                PaperRecord(**self.df.loc[j, :],
-                            record_id=j,
-                            column_spec=self.column_spec) for j in index_list
+                PaperRecord(
+                    **self.df.loc[j, :],
+                    column_spec=self.column_spec,
+                    record_id=j,
+                    record_by_id=j,
+                    record_by_index=self.reverse_index[j])
+                for j in index_list
             ]
         else:
             records = [
                 PaperRecord(**self.df.iloc[j],
                             column_spec=self.column_spec,
-                            record_id=self.df.index.values[j])
+                            record_id=j,
+                            record_by_id=self.df.index.values[j],
+                            record_by_index=j)
```

Comment:
What is the aim of passing these values?

Reply:
The point it that the user of the function can choose which kind of record_id it wants to use. Sometimes you need the internal one (record_by_index), sometimes the external one (record_by_id). And I can imagine you might not really can and you can use the default one (record_id).